javagen/src/main/java/com/azure/autorest/template/ProtocolSampleTemplate.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package com.azure.autorest.template;
+
+import com.azure.autorest.extension.base.model.codemodel.RequestParameterLocation;
+import com.azure.autorest.extension.base.plugin.JavaSettings;
+import com.azure.autorest.model.clientmodel.ClassType;
+import com.azure.autorest.model.clientmodel.ClientMethod;
+import com.azure.autorest.model.clientmodel.ClientMethodParameter;
+import com.azure.autorest.model.clientmodel.MethodGroupClient;
+import com.azure.autorest.model.clientmodel.ProtocolExample;
+import com.azure.autorest.model.clientmodel.ProxyMethodExample;
+import com.azure.autorest.model.javamodel.JavaFile;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public class ProtocolSampleTemplate implements IJavaTemplate<ProtocolExample, JavaFile> {
```

Comment:
This is too specific to LLC. Can we make it extensible to generate samples for high-level autorest generator too?

Reply:
We probably limit the scope to LLC for now.

1. For data-plane, it is usually generated code + hand-written code. Then generated sample simply does not work.
2. There is very little example in swagger for data-plane. It is useless to have good generator without good JSON.